### PR TITLE
feat(log_to_metric): dynamically convert all logs to metrics

### PR DESCRIPTION
**Pull request by dygfloyd**
enchancement(log_to_metric): add config option that will covert all logs to metrics

### CODE REVIEW

File: src/internal_events/log_to_metric.rs
Comment (pront) [inline src/internal_events/log_to_metric.rs:69]: Perhaps someone with more expertise on Vector metrics wants to review the changes to this file (cc @jszwedko)

**Review by pront — APPROVED** [review]
This MVP looks good to me. But let's also get a second review as well as a review on the introduced metrics.

**Review by hestonhoffman — APPROVED** [review]

**Review by fuchsnj — APPROVED** [review]
Overall this looks reasonable

**Comment by pront** [comment]
This is not modifying existing functionality and it a new opt-in feature. We can probably go ahead and merge, just waiting for the checks to pass 🎉

**Pull request merged**
